src/BloomBrowserUI/bookEdit/js/bloomEditing.js

```diff
@@ -1141,6 +1143,51 @@ $.fn.hasAttr = function (name) {
     return (typeof attr !== 'undefined' && attr !== false);
 };
 
+// Some custom templates have image containers embedded in bloom-editable divs, so that the text can wrap
+// around the picture. The problems is that the user can do ctrl+a, del to start over on the text, and 
+// inadvertantly remove the embedded images. So we introduced the "bloom-preventRemoval" class, and this
+// tries to safeguard element bearing that class.
+function PreventRemovalOfSomeElements(container) {
+    
+    /* this approach showed promise, but only the first time you do ctrl+all, DEL. After the undo, the bindins were not redone. 
+    $(container).find(".bloom-preventRemoval").bind("DOMNodeRemoved", function (e) {
+        alert("Removed: " + e.target.nodeName);
+        //this threw a NS_ERROR but I don't know why
+        //document.execCommand('undo', false, null);
+    });
+
+    the problem with this one is the event was raised when we weren't actually deleting it
+    $(container).bind("DOMNodeRemoved", function (e) {
+        if ($(e.target).hasClass('bloom-preventRemoval')) {
+            alert("Removed: " + e.target.nodeName);
+            document.execCommand('undo', false, null);
+        }
+        //this threw a NS_ERROR but I don't know why
+        //document.execCommand('undo', false, null);
+    });
+    */
+    //This works for the actual case that I was trying to prevent, where the user
+    //does a ctrl+all and then delete. But notice that it's not logically sound;
+    //so long as at least *one* bloom-preventRemoval child is still there, this
+    //wouldn't trigger.
+    //Note, the is event is *not* fired on the element itself in the ctrl+all, del scenario, hence
+    //the need to go up to the parent editable and attach the even their.
+    $(container).find(".bloom-preventRemoval").closest(".bloom-editable").on("input", function (e) {
+        if ($(this).find(".bloom-preventRemoval").length == 0) {
+            document.execCommand('undo');
+        }
```

Comment:
Another possibility which could be explored...but may not be worth it...would be to install a function that triggers when the selection changes, and which figures out whether any preventRemoval element is selected, and if so changes the selection to an otherwise-equivalent one that does NOT include it. (I think HTML supports discontinuous selections that would make this possible? But if not it might still be nice to shorten the selection to something they could validly delete.)